bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -860,6 +858,47 @@ void ReplicaImp::onMessage<FullCommitProofMsg>(FullCommitProofMsg *msg) {
   return;
 }
 
+void ReplicaImp::onInternalMsg(InternalMessage &&msg) {
+  metric_received_internal_msgs_.Get().Inc();
+
+  // Handle a full commit proof sent by self
+  if (auto *fcp = std::get_if<FullCommitProofMsg *>(&msg)) {
+    return onInternalMsg(*fcp);
+  }
+
+  // Handle prepare related internal messages
+  if (auto *csf = std::get_if<CombinedSigFailedInternalMsg>(&msg)) {
+    return onPrepareCombinedSigFailed(csf->seqNumber, csf->view, csf->replicasWithBadSigs);
```

Comment:
All those handlers' signatures effectively want their respective message but instead each handler takes all members of the message struct. That looks like another opportunity to refactor later.

Reply:
Yes indeed. I thought about doing that, but kept this particular change a bit smaller since it was already getting quite large. They should probably all become `onInternalMessage(SpecificMessage)`